### PR TITLE
Make Prow's deploy.sh use RBE in CI.

### PR DESCRIPTION
**Pull request by cjwagner**
/assign @fejta

### CODE REVIEW

File: prow/deploy.sh [hunk 1]
```diff
@@ -87,7 +89,7 @@ for s in {5..1}; do
     echo -n $'\r'"in $s..."
     sleep 1s
 done
-bazel run //config/prow/cluster:production.apply --platforms=@io_bazel_rules_go//go/toolchain:linux_amd64
-bazel run //config/prow/cluster/monitoring:prow_monitoring_objects.apply --platforms=@io_bazel_rules_go//go/toolchain:linux_amd64
-bazel run //config/prow/cluster/monitoring:production.apply --platforms=@io_bazel_rules_go//go/toolchain:linux_amd64
+bazel run //config/prow/cluster:production.apply --platforms=@io_bazel_rules_go//go/toolchain:linux_amd64 "${EXTRA_BAZEL_ARGS}"
```
Comment (fejta) [inline prow/deploy.sh:92]: How about `bazel run //target "$@"` and then make it `deploy.sh [--confirm] [bazel-args]`? I really dislike environment variables that create implicit deps.

Comment (cjwagner) [reply prow/deploy.sh]: Done.

**Comment by cjwagner** [comment]
Flakey as a pie
/retest

**Pull request merged**